File: core/server/project/commonGame/src/main/java/com/home/commonGame/control/PlayerVersionControl.java
package com.home.commonGame.control;

import com.home.commonBase.control.BaseVersionControl;
import com.home.commonBase.data.system.SaveVersionData;
import com.home.commonGame.part.player.Player;

/** 角色版本控制 */
public class PlayerVersionControl extends BaseVersionControl<Player>
{
	public void init()
	{
	
	}
	
	@Override
	protected SaveVersionData getVersionData(Player me)
	{
		return me.system.getPartData().version;
	}
}

File: core/client/game/src/commonGame/control/PlayerVersionControl.cs
using System;
using ShineEngine;

/// <summary>
/// 角色存储版本控制
/// </summary>
[Hotfix]
public class PlayerVersionControl:BaseVersionControl<Player>
{
	public virtual void init()
	{

	}

	protected override SaveVersionData getVersionData(Player me)
	{
		return me.system.getPartData().version;
	}
}